app/models/data_export.rb

```diff
@@ -57,15 +59,22 @@ def to_csv!
     save!
   end
 
-  private
-
-  def set_name
+  def default_name
     export_type_key = DATA_EXPORT_TYPES.invert[self.type.to_s]
-
-    self.name ||= I18n.t(
+    I18n.t(
       "data_exports.default_name",
       type: I18n.t("data_exports.types.#{export_type_key}"),
       current_time: I18n.l(Time.zone.now, format: :long)
     )
   end
+
+  private
+
+  def set_name
+    self.name = default_name if self.name.blank?
```

Comment:
Have to check blank because

a = ""
a ||= "alternative"
puts a

results in ""